sling/nlp/parser/trainer/pytorch_modules.py

```diff
@@ -640,21 +648,27 @@ def link(bldr, name, dim, cnx, prefix=True):
       else:
         raise ValueError("Unknown feature %r" % n)
 
-      name = feature.name + "/Collect"
-      collect = ff.rawop(optype="Collect", name=name)
-      collect.add_input(indices)
-      collect.add_input(activations)
-      collected = ff.var(
-          name=name + ":0", shape=[feature.num, activations.shape[1] + 1])
-      collect.add_output(collected)
+      name = feature.name + "/Gather"
+      gather = ff.rawop(optype="Gather", name=name)
+      gather.add_input(activations)
+      gather.add_input(indices)
+
+      activation_dim = activations.shape[1]
+      oov_var = ff.var(name=name + "/oov", shape=[1, activation_dim])
+      oov_var.data = feature.transform.oov.data.numpy()
+      gather.add_input(oov_var)
+
+      gathered = ff.var(name=name + ":0", shape=[feature.num, activation_dim])
+      gather.add_output(gathered)
 
-      sz = feature.transform.weight.size()
+      projection = feature.transform.projection.weight
+      sz = projection.size()
       transform = ff.var(name=feature.name + "/transform", shape=[sz[0], sz[1]])
-      transform.data = feature.transform.weight.data.numpy()
+      transform.data = projection.data.numpy()
 
       name = feature.name + "/MatMul"
       matmul = ff.rawop(optype="MatMul", name=name)
```

Comment:
Just curious... why do you use rawop for this MatMul instead of just using ff.matmul?